src/AdvApi32.Desktop/AdvApi32.Helpers.cs

```diff
@@ -14,29 +16,140 @@ namespace PInvoke
+        /// For example, "d:\\my share\\myservice.exe" should be specified as "\"d:\\my share\\myservice.exe\"".
+        /// </param>
+        /// <param name="lpServiceName">
+        /// The name of the service to install. The maximum string length is 256 characters.
+        /// The service control manager database preserves the case of the characters, but service name comparisons are always case insensitive.
+        /// Forward-slash (/) and backslash (\) are not valid service name characters.
+        /// </param>
+        /// <param name="lpDisplayName">
+        /// The display name to be used by user interface programs to identify the service.
+        /// This string has a maximum length of 256 characters. The name is case-preserved in the service control manager.
+        /// Display name comparisons are always case-insensitive.
+        /// </param>
+        /// <param name="lpDescription">
+        /// The description of the service. If this member is NULL, the description remains unchanged.
+        /// If this value is an empty string (""), the current description is deleted.
+        /// The service description must not exceed the size of a registry value of type REG_SZ.
+        /// </param>
+        /// <param name="lpServiceStartName">
+        /// The name of the account under which the service should run.
+        /// Use an account name in the form DomainName\UserName. The service process will be logged on as this user.
+        /// If the account belongs to the built-in domain, you can specify .\UserName.
+        /// </param>
+        /// <param name="lpPassword">
+        /// The password to the account name specified by the lpServiceStartName parameter.
+        /// Specify an empty string if the account has no password or if the service runs in the LocalService, NetworkService, or LocalSystem account.
+        /// If the account name specified by the <paramref name="lpServiceStartName"/> parameter is the name of a managed service account or virtual account name, the lpPassword parameter must be NULL.
+        /// </param>
+        /// <exception cref="Win32Exception">If the method fails, returning the calling thread's last-error code value.</exception>
+        /// <exception cref="ArgumentNullException"><paramref name="lpServiceName" /> or <paramref name="lpBinaryPathName"/> are NULL or empty string.</exception>
+        /// <exception cref="FileNotFoundException"><paramref name="lpBinaryPathName"/> cannot be found.</exception>
+        public static void CreateService(string lpBinaryPathName, string lpServiceName, string lpDisplayName, string lpDescription, string lpServiceStartName, string lpPassword)
         {
-            using (SafeServiceHandle hSCManager = OpenSCManager(null, null, ServiceManagerAccess.SC_MANAGER_CREATE_SERVICE))
+            if (string.IsNullOrWhiteSpace(lpBinaryPathName))
+            {
+                throw new ArgumentNullException(nameof(lpBinaryPathName));
```

Comment:
Yeah, you are right. Every time I use `String.IsNullOrWhiteSpace` method I am tempted not to do the most granular validation by using `ArgumentException` and `ArgumentNullException` accordingly.